src/c++/perf_analyzer/mock_data_loader.h

```diff
@@ -35,16 +35,29 @@ namespace triton { namespace perfanalyzer {
 ///
 class MockDataLoader : public DataLoader {
  public:
-  MockDataLoader()
+  MockDataLoader() { SetupMocks(); }
+  MockDataLoader(size_t batch_size) : DataLoader(batch_size) { SetupMocks(); }
+
+  void SetupMocks()
   {
     ON_CALL(*this, GetTotalSteps(testing::_))
         .WillByDefault([this](size_t stream_id) -> size_t {
           return this->DataLoader::GetTotalSteps(stream_id);
         });
+    EXPECT_CALL(*this, GetTotalSteps(testing::_)).Times(testing::AnyNumber());
```

Comment:
Using this convention to disable warnings might not be good practice:

https://groups.google.com/g/googlemock/c/pRyZwyWmrRE/m/dvNujGLUfaQJ

Can you remove this and make the mock a NiceMock in the test if your goal is to disable the "uninteresting" warnings?

Reply:
really annoying to have to do this in 100 different places. If there is only one test that actually cares about counting calls, why not use a StrickMock there?

Reply:
I updated MockDataLoader to be non-naggy by default (and added a naggy version)

Reply:
This solution would allow you to use `MockMyClass` in the friend declarations:

```cpp
class MockMyClass;

class MyClass {
...
  friend MockMyClass;
};

class NaggyMockMyClass : public MyClass {
...
};

class MockMyClass : public testing::NiceMock<NaggyMockMyClass> {};
```